mm2src/derives/enum_from/src/from_stringify.rs

```diff
@@ -0,0 +1,103 @@
+use crate::from_trait::get_attr_meta;
+use crate::{CompileError, IdentCtx, MacroAttr};
+use proc_macro2::{Ident, TokenStream};
+use quote::__private::ext::RepToTokensExt;
+use quote::{quote, ToTokens};
+use syn::__private::TokenStream2;
+use syn::spanned::Spanned;
+use syn::NestedMeta::Lit;
+use syn::{NestedMeta, Variant};
+
+#[derive(Debug)]
+pub(crate) enum InnerIdentTypes {
+    String,
+    Named,
+    None,
+}
+
+pub(crate) fn get_inner_ident_type(ident: Option<Ident>) -> InnerIdentTypes {
```

Comment:
I have two questions:
1. Do I understand that `EnumFromStringify` can be used this way?
```rust
enum Foo {
  #[from_stringify("String")]
  Bar(String),
}
```

2. Can we refactor `get_inner_ident_type -> InnerIdentTypes` to
```rust
/// Checks if the inner type is `String`.
fn check_inner_ident_type(ident: Option<Ident>) -> Result<(), CompileError>;
```
`EnumFromStringify` should be used in order to convert a custom type to the string only. Why do we need to handle the `Named` variant here?

Reply:
the main purpose of `get_inner_ident_type` is to check for the inner type we are converting to e.g String or the Ident itself
```rust
enum Foo {
  #[from_stringify("String")]
  Bar(String),
 #[from_stringify("Man")]
  Man(Man),
}
```

But according to your previous comment, we're allowing only types to convert `to String` now.
```rust
enum Foo {
  #[from_stringify("String", "Man")]
  Bar(String),
}
```